Maths/Prime.ts

```diff
@@ -0,0 +1,57 @@
+const assert = (condition: boolean, message: string) => {
+  if (!condition) throw Error(message);
+};
+
+/**
+ * https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes
+ * @param limit An integer _n_ > 1
+ * @returns All prime numbers from 2 through {@linkcode limit}
+ */
+export function sieveOfEratosthenes(limit: number): number[] {
+  assert(limit > 1, "limit should be an integer greater than 1");
+  assert(Number.isInteger(limit), "limit should be an integer greater than 1");
+
+  const maybePrime: boolean[] = new Array(limit + 1).fill(true);
+  for (let i = 2; i * i <= limit; i++) {
+    if (!maybePrime[i]) continue;
+    for (let j = i * i; j <= limit; j += i) {
+      maybePrime[j] = false;
+    }
+  }
+
+  return maybePrime
+    .reduce(
+      (primes, isPrime, number) => (isPrime ? [...primes, number] : primes),
```

Comment:
I'm currently finishing this PR. There is a little spelling mistake in your suggestion. In the condition of the for-loop you typed `primes` instead of `maybePrimes` :)

Reply:
> I'm currently finishing this PR. There is a little spelling mistake in your suggestion. In the condition of the for-loop you typed `primes` instead of `maybePrime` :)

Indeed, that ought to be `maybePrime`.